src/app/static/src/app/utils.ts

```diff
@@ -37,33 +38,44 @@ export const mapStatePropByName = <T>(namespace: string | null, name: string): C
     return (namespace && mapState(namespace, [name])[name]) || mapState([name])[name]
 };
 
-export const mapStateProps = <S, K extends string>(namespace: string,
-    map: Dict<(this: CustomVue, state: S) => any>) => {
-    type R = {[key in K]: any}
-    return mapState<S>(namespace, map) as R
+export const mapStateProps = <M extends Dict<(this: CustomVue, state: any) => any>>(namespace: string, map: M) => {
+    type T = typeof map
+    type R = {[key in keyof T]: ComputedGetter<ReturnType<T[key]>>}
+    return mapState(namespace, map) as R
 };
 
-export const mapGetterByName = <T>(namespace: string | null, name: string): ComputedWithType<T> => {
+export const mapRootStateProps = <M extends Dict<(this: CustomVue, state: any) => any>>(map: M) => {
+    type T = typeof map
+    type R = {[key in keyof T]: ComputedGetter<ReturnType<T[key]>>}
+    return mapState(map) as R
+};
```

Comment:
This looks fragile, does not have any return type as well. Could also include namespace.

Reply:
The return type is R in there, to satisfy typescript we need to construct the return type based on object provided to it, there is no way to know a static return type before someone uses this function. The reason this is needed is because typescript cannot extract out the keys or the return types from the functions